system_tests/logging_.py

```diff
@@ -50,8 +50,18 @@ def setUp(self):
         self.to_delete = []
 
     def tearDown(self):
+        from gcloud.exceptions import NotFound
         for doomed in self.to_delete:
-            doomed.delete()
+            backoff_intervals = [1, 2, 4, 8]
+            while True:
+                try:
+                    doomed.delete()
+                    break
+                except NotFound:
```

Comment:
> But if doomed is already deleted before the first doomed.delete() then it'll just error out 4 times.

Things in `doomed` were created by the test being torn down, which won't have deleted them already.

> As for #1657, can we hold off on a more conclusive answer before implementing?

This change addresses @munangst's theory that eventual consistency is what has been causing the 404's.  Do you have any other theory?  With this fix, I have all tests passing (again, after they began failing again after the brief happy interval earlier).